posthog/queries/query_date_range.py

```diff
@@ -174,21 +174,40 @@ def date_from(self) -> Tuple[str, Dict]:
 
         return date_from_query, date_from_param
 
-    def _get_timezone_aware_date_condition(self, operator: str, date_clause: str) -> str:
-        if self.should_round:
-            return f"AND toDateTime({self._table}timestamp, 'UTC') {operator} {self._timezone_date_clause(date_clause)}"
+    def _get_timezone_aware_date_condition(self, operator: Literal[">=", "<="]) -> str:
+        relevant_param = "date_from" if operator.startswith(">") else "date_to"
+        event_timestamp_expr = self._normalize_datetime(f"{self._table}timestamp", stored_in_utc=True)
+        date_expr = self._normalize_datetime(f"%({relevant_param})s")
+        if operator == ">=" and self.should_round:  # Round date_from to start of interval if `should_round` is true
+            date_expr = self._truncate_normalized_datetime(date_expr, self.interval_annotation)
+        return f"AND {event_timestamp_expr} {operator} {date_expr}"
+
+    @staticmethod
+    def _normalize_datetime(datetime_expr: str, *, stored_in_utc: bool = False) -> str:
+        """Return expression with datetime normalized to project timezone.
+        Use `from_utc=True` for datetimes from the database - we always store datetimes in UTC.
```

Comment:
from_utc is not relevant in this function - you maybe meant stored_in_utc?

Reply:
Yup, that's what I meant. Though actually `stored_in_utc` doesn't fully convey what's the intended way of using this. I reworked this to take either a `column` or `param` argument instead, since that's the key difference. Take a look.